domain/user/bootstrap/bootstrap.go

```diff
@@ -87,6 +89,8 @@ func AddUserWithPassword(name string, password auth.Password, access permission.
 			name, err))
 	}
 
+	state := state.NewSharedState(internaldatabase.DefaultStatementBase{})
```

Comment:
especially once `bootstrap` has to import two separate shared states, and both of them are in a 'state' package, and then we end up with cmdjujujuju.Something. :)

Reply:
We should just use the pattern I proposed last week.

```
permissionstate "github.com.com/juju/juju/domain/permission/state"
userstate "github.com/juju/juju/domain/user/state"
```

That way it's obvious about which state you're bringing in.